veth: fix report of veth peer on a different netns

ffmancera opened the pull request:
When the veth peer is on a different netns, the ifindex counting start
from "1" again. That means, if the veth has ifindex 30 one could expect
the peer will have ifindex 31 but if it is on a different netns, the
ifindex could be completetly different, e.g "2".

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

cathay4t commented:
We should skip the interface name lookup when the `IFLA_LINK_NETNSID` is defined.

ffmancera committed:
iface: add support to link_netnsid

The link_netnsid allow the user to know in which netns is located the veth
peer in case it is set.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera commented:
> We should skip the interface name lookup when the `IFLA_LINK_NETNSID` is defined.

It won't work because the LINK_NETNSID is the ID of the NETNS where the veth interface is.. not the peer. In nispor we do not have information about other netns interfaces yet. We need to support multi-netns state querying.. 

What do you think about this solution? It checks that the veth peer and the peer iface matches in name.

cathay4t commented:
The `IFLA_LINK_NETNSID` is the NSID of veth **peer**. https://www.spinics.net/lists/netdev/msg317678.html You may try `strace -s 1024 ip -d link` to get it.

ffmancera commented:
> The `IFLA_LINK_NETNSID` is the NSID of veth **peer**. https://www.spinics.net/lists/netdev/msg317678.html You may try `strace -s 1024 ip -d link` to get it.

You are right. My bad, I misread it. I added a new patch supporting netnsid. I tested it and it works as expected :)

ffmancera commented:
@cathay4t Take a look when you have time, thank you!

cathay4t commented on src/lib/ifaces/iface.rs:
```diff
@@ -224,6 +224,8 @@ pub struct Iface {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub controller_type: Option<ControllerType>,
     #[serde(skip_serializing_if = "Option::is_none")]
+    pub netnsid: Option<i32>,
```
This is `netnsid` might confused with namespace ID of this interface. Since the kernel is using `IFLA_LINK_NETNSID`, we can use `link_netnsid` as property name indicating the underline link is in another network namespace.

ffmancera replied:
Done, thank you!
